### PR TITLE
Fix 117 - Linking files within dsk

### PR DESCRIPTION
**Pull request by 6mint**
Refers to: https://github.com/atelierdisko/dsk/issues/117

### CODE REVIEW

File: internal/ddt/node_doc_transformer_test.go [hunk 1]
```diff
@@ -152,6 +153,7 @@ func TestNodeLinksKeepFragmentAndQueryFromOriginal(t *testing.T) {
 	dt, _ := NewNodeDocTransformer("/tree", "foo/bar", get, "test")
 
 	expected := map[string]string{
+		"<a href=\"./?answer=42\"></a>":      		"<a href=\"/tree/foo/bar?answer=42&v=test\" data-node=\"foo/bar\"></a>",
```
Comment (mariuswilms) [inline internal/ddt/node_doc_transformer_test.go]: When using `href="?answer=42"` is also accepted, please add an additional test for that.

File: internal/ddt/node_doc_transformer_test.go [hunk 2]
```diff
@@ -37,14 +37,15 @@ func TestNodeLinkAbsolute(t *testing.T) {
...
-		if url == "foo/bar/baz" {
+		if url == "foo/bar/baz" || url == "foo/bar" {
 			return true, &Node{root: "/tmp/xyz", Path: filepath.Join("/tmp/xyz", url)}, nil
 		}
 		return false, &Node{}, nil
 	}
 	dt, _ := NewNodeDocTransformer("/tree", "foo/bar", get, "test")
 
 	expected := map[string]string{
+		"<a href=\"./\"></a>":         "<a href=\"/tree/foo/bar?v=test\" data-node=\"foo/bar\"></a>",
```
Comment (mariuswilms) [inline internal/ddt/node_doc_transformer_test.go]: When using `href=""` is also accepted, please add an additional test for that.

Comment (mariuswilms) [reply internal/ddt/node_doc_transformer_test.go]: When using `href="."` is also accepted, please add an additional test for that.

File: internal/ddt/node_doc_transformer.go [hunk 3]
```diff
@@ -210,6 +210,8 @@ func (dt NodeDocTransformer) ProcessHTML(contents []byte) ([]byte, error) {
 //   ./bar
 //   bar
 //
+// "./" and "." will be resolved to the current node.
```
Comment (mariuswilms) [inline internal/ddt/node_doc_transformer.go]:
```suggestion
//
// While viewing the "bar" node, the following will resolve to itself:
//   ./ 
//   .
//  <empty string>
```

File: frontend/src/Link/index.js [hunk 4]
```diff
@@ -13,10 +13,11 @@ function Link(props) {
   if (!node) {
     return <a href={props.href}>{props.children}</a>;
   }
-  let hash = props.href.split('?t=')[1] || undefined;
+  let url = new URL(props.href, window.location.origin);
+  let params = Object.fromEntries(new URLSearchParams(url.search));
```
Comment (mariuswilms) [inline frontend/src/Link/index.js]: `params` is such a loaded word in React context, maybe you can find a different one.

Comment (mariuswilms) [reply frontend/src/Link/index.js]: Can you add an comment, on what is happening here and why? Without knowing about URL and friends it's not immdiately clear to me and probably others.

File: internal/ddt/node_doc_transformer_test.go (hunk 2 above)
Comment (6mint) [reply internal/ddt/node_doc_transformer_test.go]: 👍

File: internal/ddt/node_doc_transformer_test.go (hunk 1 above)
Comment (6mint) [reply internal/ddt/node_doc_transformer_test.go]: Good Idea

**Pull request merged**